src/antora-adapter.js

```diff
@@ -29,6 +29,13 @@ module.exports = (file, contentCatalog, vfs) => {
     read: (resourceId, format) => {
       const target = contentCatalog.resolveResource(resourceId, file.src)
       return target ? target.contents.toString() : baseReadFn(resourceId, format)
+    },
+    exists: (resourceId) => {
+      const target = contentCatalog.resolveResource(resourceId, file.src)
+      return target !== undefined
+    },
+    dirname: (resourceId) => {
+      return resourceId.substring(0, Math.max(resourceId.lastIndexOf('$'), resourceId.lastIndexOf('/'), resourceId.lastIndexOf('\\')) + 1)
```

Comment:
This is unreliable and we should instead use the `contentCatalog` to resolve the resource. I believe it's possible to build an absolute `dirname` path from the resource returned by Antora.

Reply:
Could be... but I think that'll require a lot more refactoring than I'm comfortable with doing in this code base I'm afraid.

Reply:
Ok, I see now. The issue is then that the file resolution is a bit all over the place. The `antora-adapter` does path resolution already, but the `node-fs` one doesn't; that's handled in other places but that breaks the resolution that's in `antora-adapter`. I guess the solution is to add resolution to `node-fs` and removing it from the other places.

Reply:
> This is unreliable and we should instead use the `contentCatalog` to resolve the resource. I believe it's possible to build an absolute `dirname` path from the resource returned by Antora.

Sorry, after two weeks of holidays I forgot how it worked, but this isn't there to create an absolute path. Instead it resolves the path from the PlantUML `!include` statement to a resourceId. It's still antora that's reading it from the catalog.

Reply:
We are using the Virtual File System abstraction to get information about a resource. When we are using this extension through Antora we need to use the content catalog.

Antora knows where a resource is located and can compute a dirname. We need an absolute diagram directory path since PlantUML `!include` relative path will be resolved using this diagram directory.


**/path/to/modules/ROOT/pages/doc.adoc**
```
plantuml::example$architecture/diagram.puml[]
```

**/path/to/modules/ROOT/examples/architecture/diagram.puml**
```
!include ../styles.puml

Alice -> Bob: Hello
```

**/path/to/modules/ROOT/examples/styles.puml**
```
skinparam monochrome true
skinparam ranksep 20
skinparam dpi 150
skinparam arrowThickness 0.7
```

Antora knows that `example$architecture/diagram.puml` is located in `/path/to/modules/ROOT/examples/architecture`. If we want to resolve `../styles.puml` we need to compute the dirname (using Antora):

```
/path/to/modules/ROOT/examples/architecture/../styles.puml -> /path/to/modules/ROOT/examples/styles.puml
```

Does it make sense? Am I missing something?

Reply:
How I read the antora-adapter is that it doesn't resolve the resource ID to a file path to be read, but it returns the actual contents of the resource: https://github.com/Mogztter/asciidoctor-kroki/blob/master/src/antora-adapter.js#L31

We can't resolve the included file relative to the diagrams file path because we don't have it; we only have the Antora resource ID which is the "example$architecture/diagram.puml", so we have to find a resource ID for the included file by getting the "dirname" from the diagram resource ID and resolving the included file through that so the content catalog can provide it.

The reason I think why it wasn't working on Windows is that the path joining used `path.join` which on Windows will use backslash instead of forward slash, and I suspect Antora doesn't like that. I've committed a change that should fix it for real this time. I don't have a running Windows dev environment to test though.